### ISSUE TITLE
Consider allowing partial subcommand matching for `dict`

### ISSUE THREAD

**Issue by nmoroze** [nmoroze/tclint#16]
I noticed the following Tcl works:

```tcl
set foo [dict create]
puts [dict exist $foo asdf]
```

even though the full command is meant to be `dict exists`. This is flagged as a `command-args` violation in `tclint` despite it being valid, I should consider whether this should remain the case. I also need to check if other built-in commands with subcommands behave the same way.

I think it's probably better style to use full command names, but it could perhaps be a separate violation to make it easier for users to waive.
